src/wallet/spend.cpp

```diff
@@ -582,7 +582,13 @@ std::optional<SelectionResult> SelectCoins(const CWallet& wallet, CoinsResult& a
     if (coin_control.HasSelected() && !coin_control.m_allow_other_inputs) {
         SelectionResult result(nTargetValue, SelectionAlgorithm::MANUAL);
         result.AddInput(preset_inputs);
-        if (result.GetSelectedValue() < nTargetValue) return std::nullopt;
+
+        if (result.UseEffectiveValue() && result.GetSelectedEffectiveValue() < nTargetValue) {
+            return std::nullopt;
+        } else if (result.GetSelectedValue() < nTargetValue) {
+            return std::nullopt;
+        }
```

Comment:
We always use the effective value of UTXOs except when the recipient pays the fees via SFFO, so you can go by the negation of the corresponding coin selection parameter:

```suggestion
        if (!coin_selection_params.m_subtract_fee_outputs && result.GetSelectedEffectiveValue() < nTargetValue) {
            return std::nullopt;
        } else if (result.GetSelectedValue() < nTargetValue) {
            return std::nullopt;
        }
```

Perhaps it would be nicer if we dropped the negation and swapped `if` and `else if` branches:

```suggestion
        if (coin_selection_params.m_subtract_fee_outputs && result.GetSelectedValue() < nTargetValue) {
            return std::nullopt;
        } else if (result.GetSelectedEffectiveValue() < nTargetValue) {
            return std::nullopt;
        }
```

or more compactly:

```suggestion
        if (result.GetSelectedEffectiveValue() < nTargetValue || coin_selection_params.m_subtract_fee_outputs && result.GetSelectedValue() < nTargetValue) {
            return std::nullopt;
        }
```

Reply:
Thanks, I pushed your first edited version for readability, the two others translate to another statement that is invalid though.